### PR TITLE
Feature/pop over tooltip

### PR DESCRIPTION
**Pull request by VictorKunzler**
Replace the changes in PR https://github.com/Codeminer42/cm42-central/pull/546.
"This PR is to add a popover component in which tells the user that when a task is accepted it can no longer be modified.
This functionality is for beta version."

image

### CODE REVIEW

File: app/assets/stylesheets/new_board/_story.scss [hunk 1]
```diff
@@ -341,11 +341,21 @@
...
+          display: block;
+
           input {
             @extend .btn;
             @extend .btn-default;
             @extend .btn-xs;
           }
+          
+          .infoToolTip {
+            background-color: #fff;
```
Comment (boniattirodrigo) [inline app/assets/stylesheets/new_board/_story.scss]: There is a file which contains all variables for colors, you can use $white here.

```suggestion
            background-color: $white;
```

File: app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js [hunk 2]
```diff
@@ -54,6 +55,17 @@ class ExpandedStoryControls extends Component {
           type="button"
           value={I18n.t('cancel')}
         />
+
+        {
+          disabled && (
+            <ExpandedStoryToolTip text={I18n.t('accepted_tooltip')}>
+              <div className="infoToolTip">
+               <i className="mi md-18">info_outline</i>
```
Comment (boniattirodrigo) [inline app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js]: You missed a space here:
```suggestion
                <i className="mi md-18">info_outline</i>
```

Make sure that you have [EditorConfig](https://editorconfig.org/) installed in your text editor/IDE.

File: app/assets/stylesheets/new_board/_story.scss [hunk 3]
```diff
@@ -341,11 +341,21 @@
         &__controls {
           @extend .btn-group;
 
+          height: 22px;
```
Comment (boniattirodrigo) [inline app/assets/stylesheets/new_board/_story.scss]: Are you sure that you need to define height here too?

File: app/assets/stylesheets/new_board/_story.scss (hunk 1 above)
Comment (gutofoletto) [reply app/assets/stylesheets/new_board/_story.scss]: do you think we really need that background? I would remove it a leave just the icon with transparent background.

File: app/assets/stylesheets/new_board/_story.scss (hunk 3 above)
Comment (VictorKunzler) [reply app/assets/stylesheets/new_board/_story.scss]: Yes, without height the div gets size 0

File: app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import ExpandedStoryToolTip from './ExpandedStoryToolTip'
```
Comment (marclerodrigues) [inline app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js]: I see that we're using semi-colons at the end of the imports. We should add it here too. :smile:

File: spec/javascripts/components/story/expanded_story/expanded_story_tooltip_spec.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import ExpandedStoryTooltip from 'components/story/ExpandedStory/ExpandedStoryToolTip';
+
+describe('<ExpandedStoryTooltip />', () => {
+  it('renders <Popover />', () => {
```
Comment (marclerodrigues) [inline spec/javascripts/components/story/expanded_story/expanded_story_tooltip_spec.js]: Shouldn't we add more specs? As far as I can see, Okay we have the component rendered, but what about what is being rendered and how it is being rendered?

Comment (paulodiovani) [reply spec/javascripts/components/story/expanded_story/expanded_story_tooltip_spec.js]: Yeah, but this component uses a `renderContent` prop, which is kinda tricky to test, so we decided to not add much complexity to the test with so small gain.

I see, however, that the coverage dropped a lot, so we're open to suggestions.

File: app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js [hunk 6]
```diff
@@ -54,6 +55,10 @@ class ExpandedStoryControls extends Component {
           type="button"
           value={I18n.t('cancel')}
         />
+
+        {
+          disabled && <ExpandedStoryToolTip text={I18n.t('accepted_tooltip')} />
```
Comment (paulodiovani) [inline app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryControls.js]: @VictorKunzler This is a good point to add a test and probably the biggest reason to coverage drop.
Please, add an extra test setting the `disabled` prop to `true`.

File: app/assets/javascripts/components/story/ExpandedStory/index.js [hunk 7]
```diff
@@ -49,6 +49,7 @@ export class ExpandedStory extends React.Component {
           onDelete={() => deleteStory(story.id, project.id)}
           canSave={Story.canSave(story)}
           canDelete={Story.canDelete(story)}
+          disabled={disabled}
```
Comment (paulodiovani) [inline app/assets/javascripts/components/story/ExpandedStory/index.js]: Optionally, we can check if this prop is being properly forwarded.

File: spec/javascripts/components/story/expanded_story/expanded_story_tooltip_spec.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+    const wrapper = shallow(<ExpandedStoryTooltip />);
```
Comment (paulodiovani) [inline spec/javascripts/components/story/expanded_story/expanded_story_tooltip_spec.js]: Please, add a `text="any text"` prop here.

**Pull request merged**